State distinguishability improvement

vprusso opened the pull request:
Closes #420

purva-thakre commented on toqito/state_opt/state_distinguishability.py:
```diff
@@ -69,61 +72,83 @@ def state_distinguishability(
+    :param probs: Respective list of probabilities each state is selected. If no
+                  probabilities are provided, a uniform probability distribution is assumed.
+    :param solver: Optimization option for `picos` solver. Default option is `solver_option="cvxopt"`.
+    :param primal_dual: Option for the optimization problem.
```
```suggestion
    :param primal_dual: Option for the optimization problem. Default option is `"dual"`.
```

purva-thakre commented on toqito/state_opt/state_distinguishability.py:
```diff
@@ -69,61 +72,83 @@ def state_distinguishability(
     >>> states = [e_00, e_11]
     >>> probs = [1 / 2, 1 / 2]
-    >>> res = state_distinguishability(states, probs)
+    >>> res, _ = state_distinguishability(states, probs)
```
I think it might be better to add a few examples that show off the different solver options: `primal_dual`. A user has the choice of `dual` or `primal`. 

This might be a better option as the private functions  `_min_error_primal` and ` _min_error_dual` are not going to show up in the docs build.

Also, for the existing example, make it obvious somehow that we are using the default option of `dual`.

vprusso replied:
Yep, added some additional details in the docstring.

purva-thakre commented on toqito/state_opt/state_distinguishability.py:
```diff
@@ -1,12 +1,15 @@
+    vectors: list[np.ndarray],
+    probs: list[float] = None,
+    solver: str = "cvxopt",
+    primal_dual: str = "dual",
```
Since you have removed `dist_method`, you should remove the options for it from the docstring. 

Stuff under `math::` will also change due to this.

vprusso replied:
Yep, good point. Changed!

purva-thakre commented on toqito/state_opt/tests/test_state_distinguishability.py:
```diff
@@ -1,92 +1,34 @@
+])
+def test_state_distinguishability_invalid_vectors(vectors, probs, solver, primal_dual):
+    """Test function works as expected for an invalid input."""
+    with pytest.raises(ValueError):
```
```suggestion
    with pytest.raises(ValueError, match = "Vectors for state distinguishability must all have the same dimension."):
```
It's better to use `match` because I have noticed instances of pytest thinking a line is uncovered because we did not make it check the error message.

vprusso commented:
Thanks again for the feedback, @purva-thakre 

I believe I have addressed your comments. I have also added a function `is_distinguishable` which functions very similarly to `is_antidistinguishable` (but checks for distinguishability over antidistinguishability) 

This should also close: https://github.com/vprusso/toqito/issues/426

purva-thakre commented on toqito/state_opt/state_distinguishability.py:
```diff
@@ -23,17 +26,8 @@ def state_distinguishability(
     Alice chooses :math:`i` with probability :math:`p_i` and creates the state :math:`\rho_i` Bob
     wants to guess which state he was given from the collection of states.
 
```
```suggestion
    Alice chooses :math:`i` with probability :math:`p_i` and creates the state :math:`\rho_i`. Bob
    wants to guess which state he was given from the collection of states.

```

purva-thakre reviewed (approved):
LGTM!
